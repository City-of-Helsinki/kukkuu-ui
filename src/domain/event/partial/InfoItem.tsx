import classNames from 'classnames';
import { FunctionComponent } from 'react';

import Text from '../../../common/components/text/Text';
import styles from './occurrenceInfo.module.scss';

export type InfoItemProps = {
  id: string;
  className?: string;
  icon: JSX.Element;
  label: string;
  description?: string;
  fullWidth?: boolean;
};

const InfoItem: FunctionComponent<InfoItemProps> = ({
  className,
  icon,
  label = '',
  description = '',
  fullWidth = false,
}) => {
  return (
    <div className={classNames(className, { [styles.fullWidth]: fullWidth })}>
      {icon}
      <div>
        <Text variant="body" className={styles.label}>
          {label}
        </Text>

        {description && (
          <Text className={styles.description}>{description}</Text>
        )}
      </div>
    </div>
  );
};

export default InfoItem;
