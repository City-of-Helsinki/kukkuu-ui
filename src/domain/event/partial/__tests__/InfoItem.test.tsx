import { customRender as render } from '../../../../common/test/customRender';
import InfoItem from '../InfoItem';

const infoItem = {
  id: 'time',
  className: 'class',
  icon: <i>iconSource</i>,
  iconAlt: 'Alt text',
  label: 'infoItemLabel',
};

it('renders snapshot correctly', () => {
  const { container } = render(<InfoItem key={1} {...infoItem} />);
  expect(container).toMatchSnapshot();
});
