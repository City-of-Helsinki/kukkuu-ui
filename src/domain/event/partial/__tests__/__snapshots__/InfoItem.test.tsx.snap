// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    class="class"
    id="time"
  >
    <i>
      iconSource
    </i>
    <div>
      <p
        class="_text_e4e173 _body_e4e173 _label_e93b69"
      >
        infoItemLabel
      </p>
    </div>
  </div>
</div>
`;
